File: Modulo01/Pontos2D-Java/Ponto2D.java
public class Ponto2D {
    
    // Atributos da classe

    private double x;
    private double y;

    public Ponto2D(double x, double y) { // Construtor
        this.x = x;
        this.y = y;
    }

    // Métodos getters and setters

    public double getX() {
        return this.x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return this.y;
    }

    public void setY(double y) {
        this.y = y;
    }

    // Outros métodos da classe

    public double distancia(Ponto2D p) {
        double dx = x - p.x;
        double dy = y - p.y;
        return Math.sqrt(dx * dx + dy * dy);
    }

    public void moveX(double dx) {
        this.x += dx;
    }
    
    public void moveY(double dy) {
        this.y += dy;
    }

    public void imprime() {
        System.out.printf("(%.2f, %.2f)\n", this.x, this.y);
    }
}

File: Modulo01/Pontos2D-Cpp/Ponto2D.cpp
#include <cmath>
#include <cstdio>
#include <iostream>
#include <iomanip>
#include "Ponto2D.h"

using namespace std;

// construtor

Ponto2D::Ponto2D(double x, double y) {
    this->x = x;
    this->y = y;
}

// destrutor

Ponto2D::~Ponto2D() {
}

// Métodos getters and setters

double Ponto2D::getX() const {
    return this->x;
}

void Ponto2D::setX (double x) {
    this->x = x;
}

double Ponto2D::getY() const {
    return this->y;
}

void Ponto2D::setY (double y) {
    this->y = y;
}

// Outros métodos da classe

double Ponto2D::distancia(const Ponto2D p) const {
    double dx = this->x - p.x;
    double dy = this->y - p.y;
    return sqrt(dx * dx + dy * dy);
}

void Ponto2D::moveX(double dx) {
    this->x += dx;
}

void Ponto2D::moveY(double dy) {
    this->y += dy;
}

void Ponto2D::imprime() {
    cout << fixed << setprecision(2);
    cout << "(" << this->x << ", " << this->y << ")" << endl;
}